server/src/test/java/keywhiz/service/permission/AlwaysAllowDelegatingPermissionCheckTest.java

```diff
@@ -0,0 +1,71 @@
+package keywhiz.service.permission;
+
+import com.codahale.metrics.MetricRegistry;
+import java.util.Objects;
+import keywhiz.service.permissions.Action;
+import keywhiz.service.permissions.AlwaysAllowDelegatingPermissionCheck;
+import keywhiz.service.permissions.KeywhizPrincipal;
+import keywhiz.service.permissions.PermissionCheckMetricsNames;
+import keywhiz.service.permissions.PermissionCheck;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+public class AlwaysAllowDelegatingPermissionCheckTest {
+
+  @Rule public MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock PermissionCheck delegate;
+
+  MetricRegistry metricRegistry;
+  AlwaysAllowDelegatingPermissionCheck alwaysAllow;
+  KeywhizPrincipal principal;
+  Objects target;
+
+  @Before
+  public void setUp() {
+    metricRegistry = new MetricRegistry();
+    alwaysAllow = new AlwaysAllowDelegatingPermissionCheck(metricRegistry, delegate);
+  }
+
+  @Test public void delegateIsAllowedReturnsTrue() {
+    when(delegate.isAllowed(any(), any(), any())).thenReturn(true);
+
+    boolean permitted = alwaysAllow.isAllowed(principal, Action.ADD, target);
+
+    assertThat(permitted);
+    assertThat(metricRegistry.histogram(PermissionCheckMetricsNames.PermissionCheckHistogram).getCount()).isEqualTo(1);
+    assertThat(metricRegistry.histogram(PermissionCheckMetricsNames.PermissionCheckHistogram).getSnapshot().getMean()).isEqualTo(1);
+  }
+
+  @Test public void delegateIsAllowedReturnsFalse() {
+    when(delegate.isAllowed(any(), any(), any())).thenReturn(false);
+
+    boolean permitted = alwaysAllow.isAllowed(principal, Action.ADD, target);
+
+    assertThat(metricRegistry.histogram(PermissionCheckMetricsNames.PermissionCheckHistogram).getCount()).isEqualTo(1);
+    assertThat(metricRegistry.histogram(PermissionCheckMetricsNames.PermissionCheckHistogram).getSnapshot().getMean()).isEqualTo(0);
```

Comment:
I've seen multiple major outages have delayed detection because someone messed up a merge and a critical metric line was erased and the metric stopped being emitted. A lot of people might turn up their noses at unit testing metrics, but if those metrics are what allows you to monitor critical behavior then I think it's also critical that they be tested, if only to confirm that they're still being emitted.